doozerlib/distgit.py

```diff
@@ -262,6 +262,11 @@ def release_is_recent(self, release):
         )
         return False
 
+    # @TODO: check with my peers why this method was not implemented in the
+    # base class. Looks like the parent knows too much about the children.
```

Comment:
Because the base class represents a dist-git, not an image and `_read_master_data` is poorly named (should probably be `_read_nvr_from_dockerfile` or similar. That function deals specifically with reading data out of the `Dockerfile` while distgits can also represent RPMs, which obviously wouldn't have a `Dockerfile`.